evm/vm/forks/sharding/vmc_handler.py

```diff
@@ -240,10 +250,14 @@ def guess_head(self, shard_id):
 
         :return: returns the hash of the guessed head collation
         """
+        # TODO: this should be change due to the performance issue
+        self.reset_shard_tracker(shard_id)
```

Comment:
Now I have removed this line.
Currently `guess_head` is changed like this https://github.com/ethereum/py-evm/pull/416/files#diff-dd73c470e87a612ab8844d87c3a5d172R229
At the end of the function I still need to clean up the logs in `shard_tracker`. Since it seems(still need confirming) [`guess_head`](https://github.com/ethereum/sharding/blob/develop/docs/doc.md#create_collation) should work only only with new arriving logs, since the last time `guess_head` is called.
In other words, after calling `guess_head`, new logs fetched from the network by `get_next_log` remaining in the shard_tracker should be cleaned, to avoid they being used in the future `guess_head`